include/fw_metadata.h

```diff
@@ -200,6 +203,42 @@ fw_firmware_info_get(u32_t firmware_address)
 }
 
 
+/* The supported offsets for the fw_info struct. */
+#define FW_INFO_OFFSET0 0x200
+#define FW_INFO_OFFSET1 0x400
+#define FW_INFO_OFFSET2 0x800
+#define FW_INFO_OFFSET_COUNT 3
+
+static const u32_t allowed_offsets[] = {FW_INFO_OFFSET0, FW_INFO_OFFSET1,
+					FW_INFO_OFFSET2};
+
+static_assert(ARRAY_SIZE(allowed_offsets) == FW_INFO_OFFSET_COUNT,
+		"Mismatch in the number of allowed offsets.");
+
+#if (FW_INFO_OFFSET_COUNT != 3) || \
+	(FW_INFO_OFFSET0 != CONFIG_FW_FIRMWARE_INFO_OFFSET && \
+	FW_INFO_OFFSET1 != CONFIG_FW_FIRMWARE_INFO_OFFSET && \
+	FW_INFO_OFFSET2 != CONFIG_FW_FIRMWARE_INFO_OFFSET)
+	#error FW_FIRMWARE_INFO_OFFSET not set to one of the allowed values.
+#endif
+
+/* Search for the firmware_info structure inside the firmware. */
+static inline const struct fw_firmware_info *
+fw_firmware_info_find(u32_t firmware_address)
```

Comment:
Should we address this whole "fw_firmware" issue while we are at it? It is starting to look very silly :D

Reply:
Haha, gladly. Suggestions are very welcome.

Reply:
So, "B0", "B1" and so on is conseptual names for the different stages in the boot chain. We can not use these for anything. 

We need some identifier for OUR B0, that is what we now call "Secure Boot" or "Bootloader", 
IMO we should create a PR cleaning this up once and for all. My proposal is NSB (no pun intended) for Nordic Secure Boot.  That way this would be `nsb_metadata.h`, `nsb_fw_info_find` and so on.

Reply:
Note that fw_metadata is its own subsys. This was done because it might eventually be used other places as well. But I definitely support the proposal to rename the `bootloader` subsys to something more specific :).

Reply:
at least every static function in the library does not need to be namespaced/prefixed to `fw_`